File: generate_icon.py
#!/usr/bin/env python3

from PIL import Image, ImageDraw, ImageFont
import math
import os

def create_water_bottle_icon(size):
    """Create water bottle icon at given size"""
    img = Image.new('RGB', (size, size))
    draw = ImageDraw.Draw(img, 'RGBA')

    scale = size / 1024.0

    # Background gradient (simplified as solid colors)
    # Top part
    draw.rectangle([(0, 0), (size, size * 0.3)], fill=(26, 128, 230))
    # Bottom part
    draw.rectangle([(0, size * 0.3), (size, size)], fill=(0, 179, 255))

    # Bottle cap
    cap_width = int(180 * scale)
    cap_height = int(80 * scale)
    cap_x = (size - cap_width) // 2
    cap_y = int(150 * scale)
    draw.rounded_rectangle(
        [(cap_x, cap_y), (cap_x + cap_width, cap_y + cap_height)],
        radius=int(15 * scale),
        fill=(255, 255, 255, 77)  # white with alpha
    )

    # Bottle neck
    neck_width = int(150 * scale)
    neck_height = int(60 * scale)
    neck_x = (size - neck_width) // 2
    neck_y = cap_y + cap_height
    draw.rectangle(
        [(neck_x, neck_y), (neck_x + neck_width, neck_y + neck_height)],
        fill=(255, 255, 255, 64)
    )

    # Main bottle body
    bottle_width = int(400 * scale)
    bottle_height = int(650 * scale)
    bottle_x = (size - bottle_width) // 2
    bottle_y = neck_y + neck_height
    draw.rounded_rectangle(
        [(bottle_x, bottle_y), (bottle_x + bottle_width, bottle_y + bottle_height)],
        radius=int(60 * scale),
        fill=(255, 255, 255, 51)
    )

    # Water fill (70% of bottle)
    water_width = int(360 * scale)
    water_height = int(455 * scale)
    water_x = (size - water_width) // 2
    water_y = bottle_y + bottle_height - water_height - int(15 * scale)
    draw.rounded_rectangle(
        [(water_x, water_y), (water_x + water_width, water_y + water_height)],
        radius=int(50 * scale),
        fill=(77, 217, 255, 217)
    )

    # Highlight on bottle
    highlight_width = int(120 * scale)
    highlight_height = int(300 * scale)
    highlight_x = bottle_x + int(50 * scale)
    highlight_y = water_y + int(50 * scale)
    draw.rounded_rectangle(
        [(highlight_x, highlight_y), (highlight_x + highlight_width, highlight_y + highlight_height)],
        radius=int(20 * scale),
        fill=(255, 255, 255, 77)
    )

    # Water surface wave (simplified)
    wave_y = water_y + int(5 * scale)
    wave_points = []
    for x in range(int(water_x), int(water_x + water_width), max(1, int(5 * scale))):
        relative_x = (x - water_x) / (50 * scale)
        sine_value = math.sin(relative_x * math.pi)
        y = wave_y + int(sine_value * 8 * scale)
        wave_points.append((x, y))

    # Draw wave as polygon
    if len(wave_points) > 2:
        wave_polygon = wave_points + [(water_x + water_width, water_y + int(50 * scale)), (water_x, water_y + int(50 * scale))]
        draw.polygon(wave_polygon, fill=(128, 255, 255, 102))

    # Draw "70%" text
    try:
        # Try to use a nice bold font
        font_size = int(140 * scale)
        try:
            # macOS system fonts
            font = ImageFont.truetype("/System/Library/Fonts/Supplemental/Arial Bold.ttf", font_size)
        except:
            try:
                font = ImageFont.truetype("/Library/Fonts/Arial Bold.ttf", font_size)
            except:
                # Fallback to default
                font = ImageFont.load_default()
    except:
        font = ImageFont.load_default()

    text = "70%"

    # Get text bounding box
    bbox = draw.textbbox((0, 0), text, font=font)
    text_width = bbox[2] - bbox[0]
    text_height = bbox[3] - bbox[1]

    text_x = (size - text_width) // 2
    text_y = (size - text_height) // 2 + int(50 * scale)

    # Draw text with shadow
    shadow_offset = max(1, int(3 * scale))
    draw.text((text_x + shadow_offset, text_y + shadow_offset), text, font=font, fill=(0, 0, 0, 77))
    draw.text((text_x, text_y), text, font=font, fill=(255, 255, 255, 255))

    return img

# Icon sizes needed
icon_configs = [
    ("icon_20x20.png", 20),
    ("icon_29x29.png", 29),
    ("icon_40x40.png", 40),
    ("icon_58x58.png", 58),
    ("icon_60x60.png", 60),
    ("icon_76x76.png", 76),
    ("icon_80x80.png", 80),
    ("icon_87x87.png", 87),
    ("icon_120x120.png", 120),
    ("icon_152x152.png", 152),
    ("icon_167x167.png", 167),
    ("icon_180x180.png", 180),
    ("icon_1024x1024.png", 1024),
]

output_dir = "/Users/chathunkurera/HydraTrack/HydraTrack/Assets.xcassets/AppIcon.appiconset"

print("Generating HydraTrack water bottle icons...")
print(f"Output directory: {output_dir}\n")

for filename, size in icon_configs:
    icon = create_water_bottle_icon(size)
    filepath = os.path.join(output_dir, filename)
    icon.save(filepath, "PNG")
    print(f"✓ Created: {filename}")

print(f"\n✅ All {len(icon_configs)} icons generated successfully!")
print(f"Icons saved to: {output_dir}")

File: generate_icon.swift
#!/usr/bin/swift

import Foundation
import CoreGraphics
import ImageIO
import UniformTypeIdentifiers

// Icon sizes needed for iOS app
let sizes: [CGFloat] = [
    20, 29, 40, 58, 60, 76, 80, 87, 120, 152, 167, 180, 1024
]

func createWaterBottleIcon(size: CGFloat) -> CGImage? {
    let colorSpace = CGColorSpaceCreateDeviceRGB()
    let bitmapInfo = CGImageAlphaInfo.premultipliedLast.rawValue

    guard let context = CGContext(
        data: nil,
        width: Int(size),
        height: Int(size),
        bitsPerComponent: 8,
        bytesPerRow: 0,
        space: colorSpace,
        bitmapInfo: bitmapInfo
    ) else { return nil }

    // Scale factor for drawing
    let scale = size / 1024.0

    // Fill background with gradient (approximation with solid color)
    context.setFillColor(red: 0.0, green: 0.7, blue: 1.0, alpha: 1.0)
    context.fill(CGRect(x: 0, y: 0, width: size, height: size))

    // Draw top gradient
    context.setFillColor(red: 0.1, green: 0.5, blue: 0.9, alpha: 1.0)
    context.fill(CGRect(x: 0, y: 0, width: size, height: size * 0.3))

    // Bottle cap
    let capWidth = 180 * scale
    let capHeight = 80 * scale
    let capX = (size - capWidth) / 2
    let capY = 150 * scale
    context.setFillColor(red: 1.0, green: 1.0, blue: 1.0, alpha: 0.3)
    let capRect = CGRect(x: capX, y: capY, width: capWidth, height: capHeight)
    let capPath = CGPath(roundedRect: capRect, cornerWidth: 15 * scale, cornerHeight: 15 * scale, transform: nil)
    context.addPath(capPath)
    context.fillPath()

    // Bottle neck
    let neckWidth = 150 * scale
    let neckHeight = 60 * scale
    let neckX = (size - neckWidth) / 2
    let neckY = capY + capHeight
    context.setFillColor(red: 1.0, green: 1.0, blue: 1.0, alpha: 0.25)
    context.fill(CGRect(x: neckX, y: neckY, width: neckWidth, height: neckHeight))

    // Main bottle body (rounded rectangle)
    let bottleWidth = 400 * scale
    let bottleHeight = 650 * scale
    let bottleX = (size - bottleWidth) / 2
    let bottleY = neckY + neckHeight
    context.setFillColor(red: 1.0, green: 1.0, blue: 1.0, alpha: 0.2)
    let bottleRect = CGRect(x: bottleX, y: bottleY, width: bottleWidth, height: bottleHeight)
    let bottlePath = CGPath(roundedRect: bottleRect, cornerWidth: 60 * scale, cornerHeight: 60 * scale, transform: nil)
    context.addPath(bottlePath)
    context.fillPath()

    // Water fill (70% of bottle)
    let waterWidth = 360 * scale
    let waterHeight = 455 * scale
    let waterX = (size - waterWidth) / 2
    let waterY = bottleY + bottleHeight - waterHeight - 15 * scale
    context.setFillColor(red: 0.3, green: 0.85, blue: 1.0, alpha: 0.85)
    let waterRect = CGRect(x: waterX, y: waterY, width: waterWidth, height: waterHeight)
    let waterPath = CGPath(roundedRect: waterRect, cornerWidth: 50 * scale, cornerHeight: 50 * scale, transform: nil)
    context.addPath(waterPath)
    context.fillPath()

    // Highlight on left side
    let highlightWidth = 120 * scale
    let highlightHeight = 300 * scale
    let highlightX = bottleX + 50 * scale
    let highlightY = waterY + 50 * scale
    context.setFillColor(red: 1.0, green: 1.0, blue: 1.0, alpha: 0.3)
    let highlightRect = CGRect(x: highlightX, y: highlightY, width: highlightWidth, height: highlightHeight)
    let highlightPath = CGPath(roundedRect: highlightRect, cornerWidth: 20 * scale, cornerHeight: 20 * scale, transform: nil)
    context.addPath(highlightPath)
    context.fillPath()

    // 70% text
    let fontSize = 140 * scale
    let text = "70%" as CFString
    let font = CTFontCreateWithName("Helvetica-Bold" as CFString, fontSize, nil)

    var attributes: [CFString: Any] = [
        kCTFontAttributeName: font,
        kCTForegroundColorAttributeName: CGColor(red: 1.0, green: 1.0, blue: 1.0, alpha: 1.0)
    ]

    let attributedString = CFAttributedStringCreate(kCFAllocatorDefault, text, attributes as CFDictionary)!
    let line = CTLineCreateWithAttributedString(attributedString)
    let bounds = CTLineGetBoundsWithOptions(line, [])

    let textX = (size - bounds.width) / 2
    let textY = (size - bounds.height) / 2 + 50 * scale

    context.textPosition = CGPoint(x: textX, y: textY)
    CTLineDraw(line, context)

    return context.makeImage()
}

func saveIconImage(_ image: CGImage, to url: URL) {
    guard let destination = CGImageDestinationCreateWithURL(url as CFURL, UTType.png.identifier as CFString, 1, nil) else {
        print("Failed to create image destination for \(url)")
        return
    }

    CGImageDestinationAddImage(destination, image, nil)

    if !CGImageDestinationFinalize(destination) {
        print("Failed to write image to \(url)")
    } else {
        print("✓ Created: \(url.lastPathComponent)")
    }
}

// Create output directory
let outputPath = "/Users/chathunkurera/HydraTrack/HydraTrack/Assets.xcassets/AppIcon.appiconset"
let outputURL = URL(fileURLWithPath: outputPath)

print("Generating HydraTrack app icons...")
print("Output directory: \(outputPath)\n")

// Generate all icon sizes
for size in sizes {
    if let image = createWaterBottleIcon(size: size) {
        let filename: String
        if size == 1024 {
            filename = "icon_1024x1024.png"
        } else {
            let scale = size.truncatingRemainder(dividingBy: 1) == 0 ? Int(size) : size
            filename = "icon_\(scale)x\(scale).png"
        }

        let fileURL = outputURL.appendingPathComponent(filename)
        saveIconImage(image, to: fileURL)
    }
}

print("\n✅ All icons generated successfully!")
print("Icons saved to: \(outputPath)")
